app/src/play/java/de/danoeh/antennapod/dialog/CustomMRControllerDialog.java

```diff
@@ -124,38 +165,89 @@ public View onCreateMediaControlView(Bundle savedInstanceState) {
-                            scale = (float) originalWidth / intrWidth;
+        LinearLayout.LayoutParams artParams;
+        if (landscape) {
+            artView = new ImageView(getContext()) {
+                @Override
+                protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+                    int desiredWidth = widthMeasureSpec;
+                    int desiredMeasureMode = MeasureSpec.getMode(heightMeasureSpec) == MeasureSpec.EXACTLY ?
+                            MeasureSpec.EXACTLY : MeasureSpec.AT_MOST;
+                    if (MeasureSpec.getMode(widthMeasureSpec) != MeasureSpec.EXACTLY) {
+                        Drawable drawable = getDrawable();
+                        if (drawable != null) {
+                            int intrHeight = drawable.getIntrinsicHeight();
+                            int intrWidth = drawable.getIntrinsicWidth();
+                            int originalHeight = MeasureSpec.getSize(heightMeasureSpec);
+                            if (intrHeight < intrWidth) {
+                                desiredWidth = MeasureSpec.makeMeasureSpec(
+                                        originalHeight, desiredMeasureMode);
+                            } else {
+                                desiredWidth = MeasureSpec.makeMeasureSpec(
+                                        Math.round((float) originalHeight * intrWidth / intrHeight),
+                                        desiredMeasureMode);
+                            }
                         }
-                        desiredHeight = MeasureSpec.makeMeasureSpec((int) (intrHeight * scale + 0.5f), MeasureSpec.EXACTLY);
                     }
+                    super.onMeasure(desiredWidth, heightMeasureSpec);
                 }
+            };
+            artParams = new LinearLayout.LayoutParams(
+                    ViewGroup.LayoutParams.WRAP_CONTENT,
+                    ViewGroup.LayoutParams.MATCH_PARENT);
+            MarginLayoutParamsCompat.setMarginStart(artParams,
+                    getContext().getResources().getDimensionPixelSize(R.dimen.media_router_controller_playback_control_horizontal_spacing));
+        } else {
+            artView = new ImageView(getContext()) {
+                @Override
+                protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+                    int desiredHeight = heightMeasureSpec;
+                    if (MeasureSpec.getMode(heightMeasureSpec) != MeasureSpec.EXACTLY) {
+                        Drawable drawable = getDrawable();
+                        if (drawable != null) {
+                            int originalWidth = MeasureSpec.getSize(widthMeasureSpec);
+                            int intrHeight = drawable.getIntrinsicHeight();
+                            int intrWidth = drawable.getIntrinsicWidth();
+                            float scale;
+                            if (intrHeight*16 > intrWidth*9) {
```

Comment:
It does not have to do with full screen or usual tv ratios. It was simply the ratio that was already being used (but not without its bugs) on the original MediaRouteControllerDialog.
I'm basically setting that, if an image is taller than 16:9, then its height should be bounded by 9/16*(available width).
Likewise, on landscape orientation, I'm making sure that if the image is horizontal, its width cannot be greater than the available height.
I saw some stuff on the webs from people trying to solve similar issues, like assigning a fixed aspect ratio to a frame layout. I'll try to find the best resource and give more meaningful comments.